test/metabase/query_processor/streaming/xlsx_test.clj

```diff
@@ -322,3 +322,11 @@
            (second (xlsx-export [{:name "val1"} {:name "val2"}]
                                 {}
                                 [[(SampleNastyClass. "Hello XLSX World!") (AnotherNastyClass. "No Encoder")]]))))))
+
+(deftest auto-sizing-threshold-test
+  (testing "Export still works when the number of rows is equal to or above the auto-sizing threshold"
+    (with-redefs [xlsx/auto-sizing-threshold 2]
+      (is (= [[1.0] [2.0]]
+             (rest (xlsx-export [{:id 0, :name "Col1"}] {} [[1.0] [2.0]]))))
+      (is (= [[1.0] [2.0] [3.0]]
+             (rest (xlsx-export [{:id 0, :name "Col1"}] {} [[1.0] [2.0] [3.0]])))))))
```

Comment:
It would be nice to have a test that actually verified the rows/columns are getting autosized the way we expected (maybe by checking `getColumnWidth`?) -- this is definitely a good test to have but it doesn't actually check that autosizing was applied. If someone accidentally breaks autosizing in the future we'd have no tests to catch that